libpod/image/image.go

```diff
@@ -0,0 +1,193 @@
+package image
+
+import (
+	"fmt"
+	"io"
+	"os"
+
+	"github.com/containers/image/docker/reference"
+	"github.com/containers/storage"
+	"github.com/pkg/errors"
+	"github.com/projectatomic/libpod/libpod"
+	"github.com/projectatomic/libpod/pkg/inspect"
+)
+
+type Image struct {
+	inspect.ImageData
+	InputName string
+	Local     bool
+	Runtime   *libpod.Runtime
+	image     *storage.Image
+}
+
+// NewFromLocal creates a new image object that is intended
+// to only deal with local images already in the store (or
+// its aliases)
+func NewFromLocal(name string, runtime *libpod.Runtime) (Image, error) {
+	image := Image{
+		InputName: name,
+		Local:     true,
+		Runtime:   runtime,
+	}
+	localImage, err := image.GetLocalImage()
+	if err != nil {
+		return Image{}, err
+	}
+	image.image = localImage
+	return image, nil
+}
+
+// New creates a new image object where the image could be local
+// or remote
+func New(name string, runtime *libpod.Runtime) (Image, error) {
+	// We don't know if the image is local or not ... check local first
+	newImage := Image{
+		InputName: name,
+		Local:     false,
+		Runtime:   runtime,
+	}
+	localImage, err := newImage.GetLocalImage()
+	if err == nil {
+		newImage.Local = true
+		newImage.image = localImage
+		return newImage, nil
+	}
+
+	// The image is not local
+	pullNames, err := newImage.createNamesToPull()
+	if err != nil {
+		return newImage, err
+	}
+	if len(pullNames) == 0 {
+		return newImage, errors.Errorf("unable to pull %s", newImage.InputName)
+	}
+	var writer io.Writer
+	writer = os.Stderr
+	for _, p := range pullNames {
+		imageName, err := newImage.pull(p, writer, runtime)
```

Comment:
Do we always want to pull here? I can see wanting to leave an image remote to do a remote inspect. Maybe a bool in there as to whether to pull the image?

Reply:
@mheon after thinking about this more, I think NewFromRemote is something we will have to end up doing.  But we can always mutate this as needed down the road.

Reply:
Alright, I'd be fine with having a separate one that keeps it remote, as long as it's clear what each does